coriolis/tests/api/v1/test_minion_pools.py

```diff
@@ -0,0 +1,431 @@
+
+                mock_validation_api_method.assert_called_once_with(
+                    ctxt,
+                    minion_pool["endpoint_id"],
+                    minion_pool["environment_options"]
+                )
+                mock__check_pool_numeric_values.assert_called_once_with(
+                    minimum_minions, maximum_minions, minion_max_idle_time)
+                mock__check_pool_retention_strategy.assert_called_once_with(
+                    minion_retention_strategy
+                )
+
+    @mock.patch.object(minion_pool_view, 'single')
+    @mock.patch.object(api.API, 'create')
+    @mock.patch.object(minion_pools.MinionPoolController,
+                       '_validate_create_body')
+    def test_create(
+        self,
+        mock_validate_create_body,
+        mock_create,
+        mock_single
+    ):
+        mock_req = mock.Mock()
+        mock_context = mock.Mock()
+        mock_req.environ = {'coriolis.context': mock_context}
+        mock_body = {}
+        mock_validate_create_body.return_value = (mock.sentinel.value,) * 11
+
+        result = self.minion_pools.create(mock_req, mock_body)
+
+        self.assertEqual(
+            mock_single.return_value,
+            result
+        )
+
+        mock_context.can.assert_called_once_with(
+            "migration:minion_pools:create")
+        mock_validate_create_body.assert_called_once_with(
+            mock_context, mock_body)
+        mock_create.assert_called_once()
+        mock_single.assert_called_once_with(mock_create.return_value)
+
+    @mock.patch.object(endpoints_api.API,
+                       'validate_endpoint_destination_minion_pool_options')
+    @mock.patch.object(endpoints_api.API,
+                       'validate_endpoint_source_minion_pool_options')
+    @ddt.file_data('data/minion_pools_validate_enviroment_options.yml')
+    def test__validate_updated_environment_options(
+        self,
+        mock_validate_endpoint_source_minion_pool_options,
+        mock_validate_endpoint_destination_minion_pool_options,
+        platform,
+        exception_raised
+    ):
+        platform = getattr(constants, platform, None)
+        endpoint_id = mock.sentinel.endpoint_id
+        minion_pool = {
+            'platform': platform,
+            'endpoint_id': endpoint_id
+        }
+        enviroment_options = mock.sentinel.enviroment_options
+        context = mock.sentinel.context
+
+        if exception_raised:
+            self.assertRaisesRegex(
+                Exception,
+                exception_raised,
+                self.minion_pools._validate_updated_environment_options,
+                context,
+                minion_pool,
+                enviroment_options
+            )
+        else:
+            self.minion_pools._validate_updated_environment_options(
+                context,
+                minion_pool,
+                enviroment_options
+            )
+
+        if platform == constants.PROVIDER_PLATFORM_SOURCE:
```

Comment:
This is not ok. I'd suggest using `getattr` to retrieve the used method from `self._endpoint_api`, and only mock that one, instead of adding this logic to the test method, similar to how you do here: https://github.com/cloudbase/coriolis/blob/dadad51ee6bd7576e4c8b96d215b71e3e85fe337/coriolis/tests/api/v1/test_migrations.py#L192-L193